src/tiledimage.js

```diff
@@ -1883,6 +1939,56 @@ function compareTiles( previousBest, tile ) {
     return previousBest;
 }
 
+/**
+ * This function checks if the path cropping is ready.
+ * @param {OpenSeadragon.TiledImage} tiledImage - tiledImage to check the cropping
+ * properties: _croppingPolygons, _croppingPaths.
+ * @param {boolean} usedClip - indicates that the canvas used clip already
+ * @return true if no condition failed, false otherwise.
+ */
+function isCroppingPathReady(tiledImage, usedClip) {
+    var flag = true;
+    // Skip checking if _croppingPath is falsely
+    if (!tiledImage._croppingPaths) {
+        return false;
+    }
+    if (usedClip) {
+        $.console.log('Unable to crop with path, canvas was clipped already.');
+        flag = false;
+    }
+    if (tiledImage._croppingPolygons && tiledImage._croppingPaths) {
+        $.console.log(
+            'TiledImage is unable crop polygons and paths at same time,' +
+            'call tileImage.resetCroppingPolygons() to crop with paths only.'
+        );
+        flag = false;
+    }
+    return flag;
+}
+
+/**
+ * This function crops the view with tiledImage._croppingPaths by transforming
+ * the canvas by the current viewport.
+ * @param {OpenSeadragon.TiledImage} tiledImage - tiledImage with _croppingPaths.
+ * @see isCroppingPathReady
+ */
+function cropContextWithCroppingPaths(tiledImage, useSketch) {
+    try {
+        var context = tiledImage._drawer._getContext(useSketch);
+        var viewport  = tiledImage._drawer.viewport;
+        var oldMatrix = context.getTransform();
```

Comment:
As noted in https://github.com/openseadragon/openseadragon/issues/1912, this won't work on IE 11, which we still support. That said, IE 11 also doesn't support Path2D, so it's actually fine to use in this context (in other words, you won't be using this feature at all on IE 11). We just need to make sure this code doesn't get hit when not using this feature (which certainly seems to be the case).